Add issue template (#2804)
patternfly/patternfly-react: .github/ISSUE_TEMPLATE/patternfly-issue-template.md

> Please populate the PatternFly issue template so we can learn more about your needs.

Reviewer: This line needs to be commented. Also, please add this codesandbox link for people to easily reproduce issues: https://codesandbox.io/s/x2isc

Author: What do you mean by it needs to be commented? It is just a general description of the template. I can add the link for sure.

Reviewer: I mean it needs to look like this: ```
<!-- Please populate the PatternFly issue template so we can learn more about your needs. Here's a codesandbox link you can reproduce your issue in: https://codesandbox.io/s/x2isc -->
```

The `<!-- -->` makes it a comment that won't appear as plain text inside the issue.

---

> name: PatternFly issue template

Reviewer 2: is this block needed?

Author: No, I don't think so.

about: Please populate the PatternFly issue template so we can learn more about your
  needs.
title: ''
labels: ''
assignees: ''

---

> 

Reviewer 2: Can we also add a question "is there a workaround?"

Author: Yes, and maybe we wrap that into @redallen 's question about it being blocking. What about "If this issue is a bug, Is this issue blocking you or is there a work around?" cc @dgutride

> ### Describe the issue. What are the expected and unexpected behavior categories?

Reviewer 2: ```suggestion
### Describe the issue. Including the expected and unexpected behavior
```

> ### Describe the issue. What are the expected and unexpected behavior categories?

Reviewer: I don't like this wording. As a dev, I'm not sure what a behaviour "category" is.

> ### Please provide the steps to reproduce so we can better understand and address the issue. Feel free to link CodePen or another tool.

Reviewer 3: Would be good to provide a ready-made environment that already pulls in PF etc

Reviewer: This is on my to-do list, especially with the -org refactor. You can add a followup issue and assign me!

### Please provide the steps to reproduce so we can better understand and address the issue. Feel free to link CodePen or another tool.

> ### Would you consider this issue a bug or enhancement?

Reviewer: I think this question should be asked first and read like "Is this a bug or feature request?"

Author: @redallen we chose this wording because we felt it was most helpful for users who may not know what a feature request is. cc @tlabaj @dgutride

Reviewer 2: Maybe "Is this a bug or an enhancement?"

> ### What is your product and what release version are you targeting?

Reviewer: Could also ask if this is blocking product work.

Author: What do you think about this feedback @tlabaj ?

Reviewer 2: I think the wording here is what the team agreed upon.  This give us more insight into an actual timeline for when the feature is needed. 
cc @LHinson
